### PR TITLE
eng-185 (#283)

### PR DESCRIPTION
**Pull request by Connorbelez**
eng-175 (#283)

### TL;DR

Added webhook infrastructure for handling payment reversals from Rotessa and Stripe, including signature verification, shared reversal processing logic, and comprehensive test coverage.

### What changed?

Added webhook endpoints `/webhooks/rotessa` and `/webhooks/stripe` to handle payment reversal events from both providers. The implementation includes:

- **Signature verification** for both Rotessa (HMAC-SHA256) and Stripe (timestamped HMAC) webhooks using Node.js crypto
- **Shared reversal processing** that looks up collection attempts, validates state, and fires GT transitions to reverse payments
- **Provider-specific handlers** that parse webhook payloads and normalize them into a common format
- **Comprehensive test suite** covering signature verification, payload mapping, reversal logic, and edge cases like duplicate webhooks and invalid states

The system processes NSF, PAD returns, disputes, and manual reversals by transitioning collection attempts from "confirmed" to "reversed" state and posting cash ledger reversal entries.

### How to test?

Run the test suite with `npm test` to verify:
- Signature verification for both providers
- Webhook payload parsing and normalization  
- Reversal processing logic including idempotency
- Edge cases like already-reversed attempts and missing collection attempts

For manual testing, configure `ROTESSA_WEBHOOK_SECRET` and `STRIPE_WEBHOOK_SECRET` environment variables and send POST requests to the webhook endpoints with properly signed payloads.

### Why make this change?

This enables automated handling of payment reversals from external providers, ensuring the system stays in sync when payments fail or are disputed. The webhook handlers provide real-time processing of NSF events, PAD returns, and disputes, automatically updating collection attempt states and posting appropriate cash ledger entries to maintain accurate financial records.

## Summary by CodeRabbit

* **New Features**
  * Transfer domain: create/initiate transfers, lifecycle state machine, provider integration (manual), and bridged-transfer bridge
  * VoPay webhook endpoint for incoming transfer status updates

* **Infrastructure**
  * Cash-ledger posting for receipts and lender payouts; transfer schema extended with idempotency, provider, and audit fields
  * Reconciliation cron with orphan detection, retry/escalation and self-healing workflows

* **Tests & Docs**
  * Extensive unit and state-machine tests; added Cash & Obligations Ledger developer guide

eng-184

### CODE REVIEW

File: specs/ENG-184/chunks/chunk-05-mutations-bridge/context.md [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+  // Map obligation type → transfer type
+  // This is a best-effort mapping — the planEntry doesn't carry obligation type directly
+  // Default to 'borrower_interest_collection' as the most common case
+  return "borrower_interest_collection";
+}
+```
+
+**Key points:**
+- The bridge creates the transfer record DIRECTLY with `status: "confirmed"` — it does NOT go through the transition engine
+- `collectionAttemptId` is set → `publishTransferConfirmed` effect will SKIP cash posting (Decision D4)
```
[the remark this answers is not in the document]
Comment (Connorbelez) [reply specs/ENG-184/chunks/chunk-05-mutations-bridge/context.md]: Fixed. The actual implementation in `convex/engine/effects/collectionAttempt.ts` (lines 163-196) already creates the bridge transfer with `status: "initiated"` and then calls `executeTransition(ctx, { entityType: "transfer", entityId: bridgeTransferId, eventType: "FUNDS_SETTLED", ... })` to reach confirmed state through the Governed Transition engine. This ensures the audit journal, hash chain entry, and effect scheduling all flow through the standard GT pipeline. The spec file's `status: "confirmed"` direct insertion was the original plan that was corrected during implementation — no code changes needed.
